docs/GETTINGSTARTED.md

```diff
@@ -0,0 +1,200 @@
+* `clusters` is a list of Kubernetes clusters, where:
+  * `token` is the Kube token of a service account that is able to get ingress resources
+  * `apiServer` is the address of the Kube api server
+  * `ca` is the Kube API CA certificate
+
+#### Kubernetes service account for Yggdrasil
+
+The `token` mentioned above must be for a service account which is able to list and get ingress resources in the Kube cluster. We can create a service account specifically for Yggdrasil with the below ClusterRole and matching ClusterRoleBinding.
+
+Create the service account:
+```console
+$ kubectl create serviceaccount yggdrasil-sa
+```
+
+Apply the following manifest for the ClusterRole:
+```yaml
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  namespace: '*'
+  name: yggdrasil-read-only
+rules:
+- apiGroups: ["extensions"]
+  resources: ["ingresses"]
+  verbs: ["get", "list", "watch"]
+```
+
+And apply the following ClusterRoleBinding:
+```yaml
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: yggdrasil-sa-binding
+subjects:
+- kind: ServiceAccount
+  name: yggdrasil-sa
+  namespace: default
+roleRef:
+  kind: ClusterRole
+  name: yggdrasil-read-only
+  apiGroup: rbac.authorization.k8s.io
+```
+
+The Yggdrasil docker container can now be started - make sure to mount the config file you have created, as well as the Kube API CA cert:
+
+```console
+$ docker run -d -v /path/to/config.yaml:/config.yaml -v /path/to/ca.crt:/ca.crt quay.io/uswitch/yggdrasil:603485bbf43b90c5fb42b743a01538ec8a4e988a --config=config.yaml --debug --upstream-port=80
+```
+
+By default, Yggdrasil will use an upstream ingress port of 443 (HTTPS), as we are just running an HTTP ingress we will use the `--upstream-port=80` flag as seen above.
+
+`Note:` For more information on Yggdrasil's flags, please see [here](/README.md#Flags).
+
+## Configure envoy
+With the Yggdrasil container running, we can now configure an envoy node. Pull an envoy v1.10 docker image with the following command:
+
+```console
+$ docker pull envoyproxy/envoy-alpine-debug:v1.10.0
+```
+
+Next, we will need to setup a minimal config file to create the admin listener for envoy, as well as pointing to our dynamic configuration provider - Yggdrasil:
+
+```yaml
+admin:
+  access_log_path: /tmp/admin_access.log
+  address:
+    socket_address: { address: 0.0.0.0, port_value: 9901 }
+
+dynamic_resources:
+  lds_config:
+    api_config_source:
+      api_type: GRPC
+      grpc_services:
+        envoy_grpc:
+          cluster_name: xds_cluster
+  cds_config:
+    api_config_source:
+      api_type: GRPC
+      grpc_services:
+        envoy_grpc:
+          cluster_name: xds_cluster
+
+static_resources:
+  clusters:
+  - name: xds_cluster
+    connect_timeout: 0.25s
+    type: STATIC
+    lb_policy: ROUND_ROBIN
+    http2_protocol_options: {}
+    hosts: [{ socket_address: { address: yggdrasil, port_value: 8080 }}]
```

Comment:
I would maybe make it more obvious that address should be the ip of the container, I guarantee someone will follow this guide and get confused why yggdrasil doesn't work